src/ServiceControl.AcceptanceTests/HeartbeatMonitoring/When_an_unmonitored_endpoint_starts_sending_heartbeats.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace ServiceBus.Management.AcceptanceTests.HeartbeatMonitoring
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Contexts;
+    using NServiceBus;
+    using NServiceBus.AcceptanceTesting;
+    using NUnit.Framework;
+    using ServiceControl.CompositeViews.Endpoints;
+    using Conventions = NServiceBus.AcceptanceTesting.Customization.Conventions;
+
+    public class When_an_unmonitored_endpoint_starts_sending_heartbeats : AcceptanceTest
+    {
+        static string EndpointName => Conventions.EndpointNamingConvention(typeof(MyEndpoint));
+
+        [Test]
+        public void Should_be_marked_as_monitored()
+        {
+            var context = new MyContext();
+            List<EndpointsView> endpoints = null;
+
+            Define(context)
+                .WithEndpoint<MyEndpoint>()
+                .Done(c => TryGetMany("/api/endpoints/", out endpoints, e => e.Name == EndpointName))
+                .Run();
+
+            var myEndpoint = endpoints.FirstOrDefault(e => e.Name == EndpointName);
+            Assert.NotNull(myEndpoint);
+            Assert.IsFalse(myEndpoint.Monitored);
+
+            Define(context)
+                .WithEndpoint<MyEndpointWithHeartbeat>()
+                .Done(c => TryGetMany("/api/endpoints/", out endpoints, e => e.Name == EndpointName && e.Monitored && e.MonitorHeartbeat && e.IsSendingHeartbeats))
+                .Run();
+
+            myEndpoint = endpoints.FirstOrDefault(e => e.Name == EndpointName);
```

Comment:
Should this be SingleOrDefault? Just in case the two runs generate two endpoint records with the same name.